pkg/controller/volume/persistentvolume/scheduler_binder.go

```diff
@@ -357,20 +423,59 @@ func (b *volumeBinder) findMatchingVolumes(pod *v1.Pod, claimsToBind []*bindingI
 		// Find a matching PV
 		bindingInfo.pv, err = findMatchingVolume(bindingInfo.pvc, allPVs, node, chosenPVs, true)
 		if err != nil {
-			return false, err
+			return false, nil, err
 		}
 		if bindingInfo.pv == nil {
 			glog.V(4).Infof("No matching volumes for PVC %q on node %q", getPVCName(bindingInfo.pvc), node.Name)
-			return false, nil
+			unboundClaims = append(unboundClaims, bindingInfo.pvc)
+			foundMatches = false
+			continue
 		}
 
 		// matching PV needs to be excluded so we don't select it again
 		chosenPVs[bindingInfo.pv.Name] = bindingInfo.pv
+		preBoundClaims = append(preBoundClaims, bindingInfo)
 	}
 
 	// Mark cache with all the matches for each PVC for this node
-	b.podBindingCache.UpdateBindings(pod, node.Name, claimsToBind)
-	glog.V(4).Infof("Found matching volumes on node %q", node.Name)
+	if len(preBoundClaims) > 0 {
+		b.podBindingCache.UpdateBindings(pod, node.Name, preBoundClaims)
+	}
+
+	if foundMatches {
+		glog.V(4).Infof("Found matching volumes for all unbound claims on node %q", node.Name)
+	}
+
+	return
+}
+
+func (b *volumeBinder) checkVolumeProvisions(pod *v1.Pod, claimsToProvision []*v1.PersistentVolumeClaim, node *v1.Node) (provisionSatisfied bool, err error) {
+	provisionedClaims := []*v1.PersistentVolumeClaim{}
+
+	for _, claim := range claimsToProvision {
+		className := v1helper.GetPersistentVolumeClaimClass(claim)
+		if className == "" {
+			return false, fmt.Errorf("no class for claim %q", getPVCName(claim))
+		}
+
```

Comment:
This should also check if "kubernetes.io/no-provisioner" is set

Reply:
Should we check if dynamic provisioning is in progress on this node? (ie selected node annotation)

Reply:
> This should also check if "kubernetes.io/no-provisioner" is set

I didn't noticed that before, so a pvc is not expected to get provisioned if the annotation is set?

> Should we check if dynamic provisioning is in progress on this node? (ie selected node annotation)

pvcs with selected node annotation should have been filtered out in `shouldDelayBinding`(called by `getPodVolumes`)

Reply:
> If the provisioner name in SC is set to "no-provisioner", then dynamic provisioning is not supported, and we should fail the predicate.

Got it.

> I think the predicate should also check for the selected node annotation, so that the scheduler does not choose a different node while provisioning is still in progress.

pvcs with selected node annotation will be filtered out in func `getPodVolumes` as type `unboundClaimsImmediate`, and the predicate will fail at the first beginning, that kind of claims will not be passed into the func here.